src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp

```diff
@@ -114,12 +127,37 @@ namespace
             return;
         }
 
-        // check if the integration is disabled
-        for (const WSTRING& disabledName : disabledIntegrationNames)
+        // ignore LoggingBuilder, it is technically non-instrumentation
+        // it is using only the same functionality
+        if (name != WStr("LoggingBuilder"))
         {
-            if (name == disabledName)
+            if (!enabledIntegrationNames.empty())
             {
-                return;
+                bool found = false;
+
+                // check if the integration is enabled
+                for (const WSTRING& enabledName : enabledIntegrationNames)
+                {
+                    if (name == enabledName)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    return;
+                }
+            }
+
+            // check if the integration is disabled
+            for (const WSTRING& disabledName : disabledIntegrationNames)
+            {
+                if (name == disabledName)
+                {
+                    return;
+                }
```

Comment:
consider extracting this code to a method that returns `true` if the instrumentation is enabled

Reply:
ce70090976a5528456e467ee52058bdc486e1531